rpcs3/Emu/Audio/Cubeb/CubebBackend.cpp

```diff
@@ -427,14 +432,14 @@ long CubebBackend::data_cb(cubeb_stream* stream, void* user_ptr, void const* /*
 
 	std::unique_lock lock(cubeb->m_cb_mutex, std::defer_lock);
 
-	if (stream != cubeb->m_stream)
+	if (!cubeb->m_reset_req.observe() && lock.try_lock_for(std::chrono::microseconds{50}) && cubeb->m_write_callback && cubeb->m_playing)
```

Comment:
Chancing a lock for a period of time needs to be well justified otherwise it looks like aworaround.

Reply:
And unlike the standard library, shared_mutex has the method is_free() which can test the ability to obtain a lock. You can write such a test before the if and call yield a single time if it's currently owned.

Reply:
It's typically taken by nonblocking code for a short period of time, so it makes sense to spin a bit.